lib/Doctrine/Migrations/Metadata/Storage/MetadataStorage.php

```diff
@@ -11,8 +11,17 @@ interface MetadataStorage
 {
     public function ensureInitialized(): void;
 
+    /**
+     * Returns only the migrations that have been successfully executed (reason=executed)
+     */
     public function getExecutedMigrations(): ExecutedMigrationsList;
 
+    /**
+     * Returns all the for which there was an execution attempt
+     * Includes skipped, errored executed migrations.
+     */
+    public function getAllExecutedMigrations(): ExecutedMigrationsList;
```

Comment:
what would be here the best way to add such functionality to this interface in a backward compatible way?

Reply:
Usually something like

```
class Foo implements MetadataStorage, AllExecutedMigrationsInterfaceThingy
```

but I assume you're looking for something different?